### PR TITLE
Added a Use Table dialog

### LINKED ISSUE #8081: Add a Use Table dialogue?
@lilyclements and @anastasia-mbithe I am really pleased with the additions to the display of tables you are working on.  Having well formatted tables is key to our reproducible work and so much easier now that we are in control of the html output ourselves.

This issue is to raise two more aspects.  - now just this one, as it has become larger!
1) I note that gt can export rtf, latex, word as well as html.  I note that the input to these options is a gt table object and we are working on saving objects.  I wonder how easy this might be.  It may become a second dialogue?  We have a Use Graph dialogue and perhaps we need a Use Table one?  

This could become a simple adaptation of the modelling idea too where we have Fit Model and then Use Model.?

As I look in the gt guide I note that quite a number of the functions assume a gr table as the first argument, so I am warmin g to the task of proposing a Use Table dialogue.  I suggest it will probably call the same Format Table sub-dialogue as one option, but might have some key aspects on the main dialogue - like the different export type perhaps?

### PR DESCRIPTION
**Pull request by anastasia-mbithe**
Fixes #8081 

This is almost ready. 
However, @Patowhiz the objects produced as either raw html, latex, word, or rtf are not supported in the output window. Is there a way we could save them as files (maybe)?

@rdstern, I have linked the dialog with the Format Options sub dialog, more edits will have to be done later when we adopt the new Format Options sub dialog, implemented in #8027

### CODE REVIEW

**Comment by rdstern** [comment]
@anastasia-mbithe I couldn't get it to work.
I first got a table.  Then opened this dialogue and tried to just then say ok to get thge table again.  It didn't give an error, but it didn't give the table either.
Then I added a title and now it gives an error and says that data is not provided.

Incidentally I tried changing the order of the summaries on a summary table.  It first didn't work.  Then when I changed them again, it gave an error.  Perhaps that isn't sorted in this pull request.

**Comment by anastasia-mbithe** [comment]
@rdstern, I have been using the script window to test this. And we dont have a table output at the moment since we changing / saving the gt tables as either rtf, word, latex or raw html formats, which are not supported by the save control right now. Thats what @Patowhiz had to look at. Sending the code to the script window and assigning the last line to a variable name i.e use_last_table i get this:
image
image

Alternatively, we can have a checkbox so that the table(s) are not saved in a certain format by default. When unchecked, we just use the table and add, maybe the formats. if checked, we save it in a certain format rtf,word,latex or raw html.

**Comment by N-thony** [comment]
@rdstern @anastasia-mbithe I think for some function which is producing the output in the format not supported yet, we can customized those function in the way to return the result in a temporary file in the format required and save it? I think this can be used as an alternative. 
We just need to sink the output to the temp file with the format requested. I can discuss more about this approach with @Patowhiz and see if it is usefull?

**Comment by rdstern** [comment]
@anastasia-mbithe and @N-thony I am happy if the different format options are disabled for now.
What I would like to be able to do is get the object back, and then change or add some of the format table options.
Later a few - perhaps some default themes, and addition of a title may be also on the main page.

**Comment by anastasia-mbithe** [comment]
@rdstern, which default themes do you suggest to go to the main page? And should we also add a title to the main page when we have it in the Format options sub dialog as well?

**Review by rdstern — COMMENTED** [review]
@anastasia-mbithe I have the same problems as when I looked before.
I used the survey data and did a minimal frequency table.  Then, in Use Table I added a title and got this error:
image


I have a zero success record.  I have not been able to change anything in a table so far, without getting this sort of error.

**Comment by anastasia-mbithe** [comment]
@rdstern , the table object to be used has to be specifically a **gt table**, that is  only created from the Tables > Frequency/Summary Tables dialog

**Comment by rdstern** [comment]
@anastasia-mbithe thanks.  I did save it from the Describe > Tables... dialogue.  So it should have been a gt table.  It was a simple 2-way frequerncy table from our little survey data of village by variety.

**Comment by rdstern** [comment]
@anastasia-mbithe my mistake as usual.  I keep forgetting to synch the master.  Now I am getting your improvements and they are exciting - but see below - again.  It is the same error as before:

image

I did multiple saving and it started well.  Very well.

Then, as you see on the bottom right, above I saved the saved one (which had a title and a footnote) and then changed the footnote.  Now it gave me 1,1 and 2,2 after the titles.  And when I deleted the footnotes, the 1 and 2 at the titles were still there.  Then I added the fottnote again and got the error shown.

**Comment by anastasia-mbithe** [comment]
@rdstern , I hope that bug is fixed now

**Comment by lilyclements** [comment]
@anastasia-mbithe this is looking really nice, and making some great headway on it. After our conversation on the "undo"-ing of changes to the table, I have some suggestions.

We want to set bits to be `NULL` if the user **removes text from a text input box**:

```
# Creating table and mmtable2
summary_table <- data_book$summary_table(data_name="Sheet1", columns_to_summarise=c("size","fert"),
                                         treat_columns_as_factor=TRUE, summaries=c("summary_mean"))
last_table <- (mmtable2::mmtable(data=summary_table, cells=value) +
                 mmtable2::header_top_left(variable='variable') + mmtable2::header_top_left(variable='summary'))

# Our table
last_table

# Then we add a title
last_table_title <- last_table %>% gt::tab_header(title="hi") %>% gt::tab_style(style=gt::cell_text(font="NULL", size=gt::px(18)), location=gt::cells_title(groups="title"))
last_table_title

# Then if we remove the title we just want to say `title = NULL`.
last_table_no_title <- last_table_title %>% gt::tab_header(title=NULL) %>% gt::tab_style(style=gt::cell_text(font="NULL", size=gt::px(18)), location=gt::cells_title(groups="title"))
# or this also works, whichever is easier:
last_table_no_title <- last_table_title %>% gt::tab_header(title=NULL)

# then we can see our last_table is the same as our table with the title removed
last_table
last_table_no_title
last_table_title
```

**Comment by rdstern** [comment]
@anastasia-mbithe and @lilyclements I assume this may now also change with the dropping of mmtable2.  Or does it only make use of gt?  Is it ready for testing again yet?

**Comment by anastasia-mbithe** [comment]
@lilyclements and @rdstern, this is ready for review.
Since, we moved from mmtable2 to gt, the functions in the sub dialog are now compatible, after making some changes, and the issues we had of repetition and not undoing are fixed now.

**Review by rdstern — APPROVED** [review]
I am approving, so we implement this dialogue (without mmtable2) as soon as possible.  @lloyddewit I hope it is ok?

Then could @anastasia-mbithe and @lilyclements also consider the items below and - if feasible - then make a new issue, or have this pull request only partially completing the existing issue?

I would like us also to reflect on what else might be possible when we use a table.
a) Could the Table to Use label be just Table.
b) Could we have a table witout margins and add, or change the margins here?  If so, then we could include the margins stuff from the main tables dialogue.
c) Could we change the layout?  If so, then we could add the layout controls from the main dialogue.  
d) If b) and c) then maybe each would start with the existing settings in the existing table?

**Comment by lilyclements** [comment]
@rdstern I have added your commented and replied to them in issue #8339

File: instat/dlgUseTable.vb [hunk 1]
```diff
@@ -16,7 +16,222 @@
...
+                                        clsNewPipeOperator:=clsPipeOperator, clsNewFootnoteTitleLocationFunction:=clsFootnoteTitleLocationFunction, clsNewFootnoteCellBodyFunction:=clsFootnoteCellBodyFunction,
+                                        clsNewFootnoteSubtitleLocationFunction:=clsFootnoteSubtitleLocationFunction, clsNewTabFootnoteSubtitleFunction:=clsTabFootnoteSubtitleFunction, clsNewJoiningOperator:=clsJoiningPipeOperator,
+                                        clsNewMutableOPerator:=clsSummaryOperator, clsNewSecondFootnoteCellFunction:=clsSecondFootnoteCellFunction,
+                                        clsNewTabStyleCellTextFunction:=clsTabStyleCellTextFunction, clsNewTabStyleFunction:=clsTabStyleFunction, clsNewTabStylePxFunction:=clsTabStylePxFunction,
+                                        clsNewgtExtraThemesFunction:=clsgtExtraThemesFunction, clsNewThemesTabOptionFunction:=clsThemesTabOptionsFunction, bReset:=bReset)
+        sdgFormatSummaryTables.ShowDialog()
+    End Sub
+    Private Sub ucrCoreControls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrTablesReceiver.ControlContentsChanged, ucrSaveTable.ControlContentsChanged
```
Comment (lloyddewit) [inline instat/dlgUseTable.vb:222]:
```suggestion

    Private Sub ucrCoreControls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrTablesReceiver.ControlContentsChanged, ucrSaveTable.ControlContentsChanged
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]
@anastasia-mbithe clean, readable code - thank you

**Pull request merged**